.github/scripts/setup_runners.sh

```diff
@@ -119,10 +121,15 @@ then
                          --parameters '{"commands": ["#!/bin/bash", "sudo hostnamectl set-hostname '${instance_name2}'"]}' \
                          --output text > /dev/null 2>&1; echo $?
 
+    echo "[${instance_name2}]: Get latest runner binary to server ${RUNNER_URL}"
+    aws ssm send-command --targets Key=tag:Name,Values=$instance_name2 --document-name "AWS-RunShellScript" \
+                         --parameters '{"commands": ["#!/bin/bash", "sudo su - '${SETUP_AMI_USER}' -c \"mkdir -p '${RUNNER_DIR}' && cd '${RUNNER_DIR}' && wget -q '${RUNNER_URL}' && tar -xzf *.tar.gz && rm *.tar.gz \""]}' \
+                         --output text > /dev/null 2>&1; echo $?
+
     echo "[${instance_name2}]: Get runner token and bootstrap on Git"
     instance_runner_token=`curl --silent -H "Authorization: token ${SETUP_TOKEN}" --request POST "${GIT_URL_API}/${GIT_URL_REPO}/actions/runners/registration-token" | jq -r .token`
     aws ssm send-command --targets Key=tag:Name,Values=$instance_name2 --document-name "AWS-RunShellScript" \
-                         --parameters '{"commands": ["#!/bin/bash", "sudo su - '${SETUP_AMI_USER}' -c \"cd actions-runner && ./config.sh --unattended --url '${GIT_URL_BASE}/${GIT_URL_REPO}' --labels '${instance_name2}' --token '${instance_runner_token}' && nohup ./run.sh &\""]}' \
+                         --parameters '{"commands": ["#!/bin/bash", "sudo su - '${SETUP_AMI_USER}' -c \"cd '${RUNNER_DIR}' && sleep 10 && ./config.sh --unattended --url '${GIT_URL_BASE}/${GIT_URL_REPO}' --labels '${instance_name2}' --token '${instance_runner_token}' && nohup ./run.sh &\""]}' \
```

Comment:
Why do you need the "sleep 10"? What is it waiting for?

Reply:
aws ssm send-command will execute the commands immediately in a standalone shell.
If you run multiple aws ssm send-command like I do, you will notice the wget/untar function above is nearly parallel with the ./config.sh & ./run.sh command in 2 separate shell.
This means, it has a possibility to execute ./config.sh & ./run.sh before wget/untar even complete. Thus, cause the script not found error and quit the run.
It is probably better to combine both commands, but that would cause the line to be too long.
If you are ok with that I can combine them for ultimate safety.

Reply:
Got it. Can you add a comment in the script explaining that?

Reply:
nvm, I found a safer way, will commit again.